components/Workspaces/WorkspacesSidebar/index.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  Box,
+  Flex,
+  HStack,
+  IconButton,
+  Link as CLink,
+  Text,
+  useColorModeValue,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { AddIcon } from '@chakra-ui/icons';
+import Link from 'next/link';
+import { useQuery } from '@apollo/client';
+import { useRouter } from 'next/router';
+import { UserQueryResponse, USER_QUERY } from '@/queries/userQuery';
+
+import { SkeletonList } from '../../SkeletonList';
+import AddWorkspaceModal from '../AddWorkspaceModal';
+
+const WorkspacesSidebar: React.FC = () => {
+  const { data, loading } = useQuery<UserQueryResponse>(USER_QUERY);
+  const { query } = useRouter();
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const background = useColorModeValue('gray.200', 'gray.600');
+
+  return (
+    <Box
+      w="256px"
+      borderRight="1px"
+      borderRightColor={background}
+      h="full"
+      pt={5}
+    >
+      <AddWorkspaceModal isOpen={isOpen} onClose={onClose} />
+      <HStack justifyContent="space-between" py={5} px={5}>
+        <Text fontWeight="semibold" fontSize="md">
+          Workspaces
+        </Text>
+        <IconButton
+          icon={<AddIcon />}
+          aria-label="Create Workspace"
+          size="xs"
+          variant="outline"
+          colorScheme="teal"
+          onClick={onOpen}
+        ></IconButton>
+      </HStack>
+      {loading ? (
```

Comment:
just a question on the theory side here - you know I'm a big fan of guards - is there a way to wrap this in a HOC or something generic to say `<ShowList loading={loading} defaultListSize={5} listComponent={ListComponent} listData={data} />` ?  Just a thought - seems like something that could remove ternaries if it got used a lot. Definitely not thinking this is necessary right now

Reply:
I'm not sure a hoc is what we're looking for here since we're describing fine grain loaders within certain sections of our components, but we can probably come up with something. For this one here we can probably just pass in `isLoading` to the skeleton stack and remove the ternary